crates/bevy_render/src/extract_component.rs

```diff
@@ -205,6 +205,8 @@ fn extract_components<C: ExtractComponent>(
     for (entity, query_item) in &query {
         if let Some(component) = C::extract_component(query_item) {
             values.push((entity, component));
+        } else {
+            commands.entity(entity).remove::<C>();
```

Comment:
@alice-i-cecile What is the behavior of remove() when the entity lacks the component already? No-op, or panic?

The docs don't say: https://dev-docs.bevyengine.org/bevy_ecs/system/commands/struct.EntityCommands.html#method.remove

Reply:
No-op.